internal/agent/agent_processor.go

```diff
@@ -66,178 +63,94 @@ func newAgentProcessor(conf *service.ParsedConfig, res *service.Resources) (serv
 	if err != nil {
 		return nil, err
 	}
-	// TODO: This sort of all seems like junk. We should write our own plugin mechanism.
-	c := exec.Command(cmd[0], cmd[1:]...)
-	c.Dir = cwd
-	c.Env = append(os.Environ(), "REDPANDA_CONNECT_AGENT_RUNTIME_MCP_SERVER="+mcpServerAddress)
-	client := plugin.NewClient(&plugin.ClientConfig{
-		HandshakeConfig:  handshake,
-		Plugins:          pluginMap,
-		Cmd:              c,
-		AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
-		SkipHostEnv:      true, // We do this ourselves
-		Logger:           &hclogger{res.Logger()},
-	})
-	rpc, err := client.Client()
+
+	// TODO: Remove this junk compatibility with the hashicorp plugin stuff, and instead
```

Comment:
Is this still up to date? I'm not sure what needs to be removed here

Reply:
Yeah it's gone now :)